jest.config-ts.json

```diff
@@ -2,6 +2,9 @@
   "clearMocks": true,
   "collectCoverageFrom": ["<rootDir>/src/**/*.ts"],
   "coveragePathIgnorePatterns": ["__tests__"],
+  "testMatch": ["<rootDir>/src/__tests__/**/**.test.ts"],
   "setupFilesAfterEnv": ["../../jest.setup.js"],
-  "testMatch": ["<rootDir>/src/__tests__/**/*.test.ts"]
+  "transformIgnorePatterns": ["/node_modules/"],
+  "testEnvironment": "node",
+  "testTimeout": 30000
```

Comment:
This comment has been updated to take into account the latest information collected by test runs. Originally I had the suspicion that tests may not run in parallel in our CI/CD, but it seems that they do

One of jest's selling points is it's ability to run tests in parallel. However, this may cause concurrency issues when using the same database for all tests.

Ideally we should

use a different database per test run thread
do not use the local dev database for the tests, but rather use an empty db (perhaps pre-populated with test-specific data)
instead of executing expensive delete statements to clean up the db, truncate tables between tests when possible
Until the above are implemented, we need to update the test timeout. I had done a bit of investigation in the past; my conclusion was that timeouts are probably related to acquiring DB connections, but I didn't figure out the exact mechanism. We also use a (more generous - 100 secs) test timeout in mocha tests. It seems that this is an old config but it did not make its way to jest when we migrated the tests. This config is also used in Tamanu (30 secs used there)

Reply:
Good point @kael89 about a more correct mechanism for using a db while testing! My concerns might be that it may become tricky to keep the testDB and actualDB in sync (in our local).

I suppose another option is for us to 'unit'-y in our unit tests. I don't really like this suggestion of mine, as I personally feel we get a lot more bang for our buck with these 'integration test style unit tests'. However at the same time they don't exactly feel like the best fit for jest? Keen to hear your thoughts on this!

Reply:
I also thought the same, but was a bit sceptical that such a popular framework like `jest` does not properly support integration tests by default. But then it dawned to me - we are probably doing DB testing wrong!

I believe that Tamanu does use a different DB per thread pattern - if my understanding is correct, when the tests run there are 3 local SQLite test dbs